src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt

```diff
@@ -12,13 +12,51 @@ Microsoft.CodeAnalysis.CSharp.DeconstructionInfo.Conversion.get -> Microsoft.Cod
 Microsoft.CodeAnalysis.CSharp.DeconstructionInfo.Method.get -> Microsoft.CodeAnalysis.IMethodSymbol
 Microsoft.CodeAnalysis.CSharp.DeconstructionInfo.Nested.get -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.DeconstructionInfo>
 Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp7_3 = 703 -> Microsoft.CodeAnalysis.CSharp.LanguageVersion
+Microsoft.CodeAnalysis.CSharp.Syntax.AnonymousFunctionExpressionSyntax.WithAsyncKeyword(Microsoft.CodeAnalysis.SyntaxToken asyncKeyword) -> Microsoft.CodeAnalysis.CSharp.Syntax.AnonymousFunctionExpressionSyntax
```

Comment:
I took a quick look at the PR and I can see a possible drawback to this change: 
1. We start with `ChildSyntax1` and `ChildSyntax2`, which both derive from `BaseSyntax` and have a `Name` field.
2. We ship this API
3. We later add a `ChildSyntax3` which also derives from `BaseSyntax`, but does not have a `Name`

We could either:
1. add a dummy `Name` to `ChildSyntax3` (so that `BaseSyntax` remains its shipped API),
2. have `ChildSyntax` should not derive from `BaseSyntax`, but rather a new `GrandBaseSyntax`
3. manually implement `BaseSyntax` APIs related to `Name` to maintain the shipped API (presumably they would throw on the new `ChildSyntax3` kind)

Alternatively to the proposed design, we could keep the current code generation design and move the `Name` field up to `BaseSyntax` (if appropriate). This must be a deliberate design decision, since it would suffer the above problem.

Tagging @gafter @jaredpar for input, since affecting public APIs.

Reply:
@jcouv Doesn't that problem exist already? From what I understand, AnonymousFunctionExpressionSyntax already has AsyncKeyword so it would make sense to have WithAsyncKeyword as well. Similarly with other base classes that have a property but not the With method.

Reply:
@jcouv Any other concerns?

Reply:
I'm waiting for @gafter to be back in office to discuss.

Reply:
great, thanks!